### PR TITLE
[benchmark] Add argument to filter simple patterns

### PR DESCRIPTION
**Pull request by chrisvittal**
used similarly to pytest -k

### CODE REVIEW

File: hail/python/hailtop/hailctl/dev/benchmark/utils.py [hunk 1]
```diff
@@ -112,10 +112,21 @@ def run_all():
...
 
+def run_pattern(pat):
+    _ensure_initialized()
+    test_run = False
+    for name, benchmark in _registry.items():
+        if pat in name:
+            test_run = True
+            _run(benchmark, _n_iter)
+    if not test_run:
+        raise ValueError(f'pattern {repr(pat)} matched no benchmarks')
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/dev/benchmark/utils.py]: not enough to request changes. but `{pat!r}` is preferred f-string syntax for repr.

File: hail/python/hailtop/hailctl/dev/benchmark/cli.py [hunk 2]
```diff
@@ -14,6 +14,8 @@ def main(args_):
                         type=int,
                         default=1,
                         help='Number of cores to use.')
+    parser.add_argument('-k', type=str, required=False, metavar='PATTERN', dest='pattern',
```
Comment (tpoterba) [inline hail/python/hailtop/hailctl/dev/benchmark/cli.py]: why `-k`? If you give this a full name `--pattern` you won't need `dest`, and it'll be consistent with the rest of the code.

File: hail/python/hailtop/hailctl/dev/benchmark/utils.py (hunk 1 above)
Comment (tpoterba) [inline hail/python/hailtop/hailctl/dev/benchmark/utils.py]: not enough to block on this alone. but `{pat!r}` is preferred f-string syntax for repr.

File: hail/python/hailtop/hailctl/dev/benchmark/cli.py (hunk 2 above)
Comment (chrisvittal) [reply hail/python/hailtop/hailctl/dev/benchmark/cli.py]: consistency with pytest, I can add `--pattern` as well.

Comment (tpoterba) [reply hail/python/hailtop/hailctl/dev/benchmark/cli.py]: ah, got it

**Pull request merged**